frontend/src/features/leases/detail/LeasePages/deposits/modal/returnedDepositModal/ReturnDepositForm.tsx

```diff
@@ -0,0 +1,119 @@
+import { FastCurrencyInput } from 'components/common/form';
+import { InlineFastDatePicker } from 'components/common/form/styles';
+import { Formik, FormikProps } from 'formik';
+import { FormLeaseDepositReturn } from 'interfaces';
+import * as React from 'react';
+import { Col, Row } from 'react-bootstrap';
+import styled from 'styled-components';
+import { formatMoney } from 'utils';
+
+import { ReturnDepositYupSchema } from './ReturnDepositYupSchema';
+
+export interface IReturnDepositFormProps {
+  formikRef: React.Ref<FormikProps<FormLeaseDepositReturn>>;
+  onSave: (values: FormLeaseDepositReturn) => void;
+  initialValues: FormLeaseDepositReturn;
+}
+
+/**
+ * Internal Form intended to be displayed within a modal window.
+ * @param {IReturnDepositFormProps} props
+ */
+export const ReturnDepositForm: React.FunctionComponent<IReturnDepositFormProps> = ({
+  initialValues,
+  formikRef,
+  onSave,
+}) => {
+  const typeDescription =
+    initialValues.depositTypeCode === 'OTHER'
+      ? 'Other - ' + initialValues.parentDepositOtherDescription
+      : initialValues.depositTypeDescription;
+
+  return (
+    <Formik
+      innerRef={formikRef}
+      enableReinitialize
+      validationSchema={ReturnDepositYupSchema}
+      onSubmit={values => {
+        onSave(values);
+      }}
+      initialValues={initialValues}
+    >
+      {formikProps => (
+        <>
+          <SubHeaderSection className="py-3 mb-4 pl-3">
```

Comment:
I don't love the mix of bootstrap classnames and styled components.

Reply:
we should chat some more about this. Our styling is not consistent atm